src/ClientHandle.cpp

```diff
@@ -3135,6 +3135,43 @@ void cClientHandle::SendUseBed(const cEntity & a_Entity, int a_BlockX, int a_Blo
 
 
 
+void cClientHandle::SendUnlockRecipe(int RecipeID)
+{
+	m_Protocol->SendUnlockRecipe(RecipeID);
+}
+
+
+
+
+
+void cClientHandle::SendInitRecipes(int RecipeID)
+{
+	m_Protocol->SendInitRecipes(RecipeID);
+}
+
+
+
+
+
+void cClientHandle::SendPrepareRecipe(int RecipeID)
+{
+	cCraftingRecipes::cRecipe* Recipe = cRoot::Get()->GetCraftingRecipes()->findRecipeById(RecipeID);
+	LOGINFO("Found Recipe %d", Recipe->m_ID);
+	for (cCraftingRecipes::cRecipeSlots::const_iterator itr = Recipe->m_Ingredients.begin(); itr != Recipe->m_Ingredients.end(); ++itr)
+	{
+		// TODO this needs to depend on the size of the grid (crafting table 3x3, just crafting 2x2)
+		LOGINFO("Ingredient %d %d %d %d", itr->x, itr->y, itr->m_Item.m_ItemType, 1 + itr->x + 2 * itr->y);
+		int RemovedItems = m_Player->GetInventory().RemoveItem(itr->m_Item);
+		this->SendWholeInventory(*m_Player->GetWindow());
+		LOGINFO("Needed %d removed %d", itr->m_Item.m_ItemCount, RemovedItems);
+		m_Player->GetWindow()->SetSlot(*m_Player, 1 + itr->x + 3 * itr->y, itr->m_Item);
+	}
+}
```

Comment:
This function is completely mis-named. I have no idea what it does, but I'm pretty sure it doesn't send anything, and it seems to remove from player inventory instead.